lib/iris/coords.py

```diff
@@ -1606,54 +1703,48 @@ def points(self, points):
         # than the desired (1,).
         if is_lazy_data(points):
             if points.shape == ():
-                points = da.reshape(points, (1,))
+                points = points.reshape(1,)
         else:
             points = self._sanitise_array(points, 1)
-        # If points are already defined for this coordinate,
-        if hasattr(self, '_points') and self._points is not None:
-            # Check that setting these points wouldn't change self.shape
-            if points.shape != self.shape:
-                raise ValueError("New points shape must match existing points "
-                                 "shape.")
 
-        self._points = points
+        # Set or update DataManager.
+        if self._points_dm is None:
+            self._points_dm = DataManager(points)
+        else:
+            self._points_dm.data = points
 
     @property
     def bounds(self):
         """
-        Property containing the bound values, as a numpy array,
+        Property containing the bound values, as a NumPy array,
         or None if no bound values are defined.
 
         .. note:: The shape of the bound array should be: ``points.shape +
             (n_bounds, )``.
 
         """
-        if self._bounds is not None:
-            bounds = self._bounds
-            if is_lazy_data(bounds):
-                bounds = as_concrete_data(bounds,
-                                          nans_replacement=np.ma.masked)
-                # NOTE: we probably don't fully support for masked aux-coords.
-                # We certainly *don't* handle a _FillValue attribute (and
-                # possibly the loader will throw one away ?)
-                self._bounds = bounds
-            bounds = bounds
-        else:
-            bounds = None
-
+        bounds = None
+        if self.has_bounds():
+            bounds = self._bounds_dm.data
         return bounds
 
     @bounds.setter
     def bounds(self, bounds):
         # Ensure the bounds are a compatible shape.
-        if bounds is not None:
-            if not is_lazy_data(bounds):
+        if bounds is None:
+            self._bounds_dm = None
+        else:
+            if not self.has_lazy_bounds():
```

Comment:
I think there is a problem here that prevents us having lazy bounds at the moment
(in at least some cases ??)
You can only use `_sanitise_arrray` on *real* data, but this is applied to lazy data if existing bounds are real, or absent.